### PR TITLE
feat(api, docs): add GPIO rail lights and door control to ProtocolContext

### LINKED ISSUE #5703: bug: set_rail_lights() not working in v3.17.1
# Overview
After updating from 3.16.1 to 3.17.1, `opentrons.drivers.rpi_drivers.gpio.set_rail_lights()` does not turn on the OT-2 rail lights and appears to be broken.

# Current behavior
Uploading a protocol with this function in the app returns the following error:
```
Your protocol could not be opened.
AttributeError [line 7]: module 'opentrons.drivers.rpi_drivers.gpio' has no attribute 'set_rail_lights'
```

# Steps to reproduce
Upload a protocol with the following code to a robot with v3.17.1 robot server installed
```
from opentrons import protocol_api
from opentrons.drivers.rpi_drivers import gpio

metadata = {'apiLevel':'2.0'}

def run(ctx):
    light = gpio.set_rail_lights(True)
    tiprack = ctx.load_labware('opentrons_96_tiprack_300ul', 1)
    plate = ctx.load_labware('corning_96_wellplate_360ul_flat', 2)
    pipette = ctx.load_instrument('p300_single_gen2', mount='right', tip_racks=[tiprack])

    pipette.transfer(300, plate['A1'], plate['A2'])
```

# Expected behavior
The expected behavior is for the code to successfully upload and turn on the rail lights during upload (protocol simulation), if the rail lights are turned off before prompting the upload. Note that this occurs successfully on v3.16.1 robot server version.

### PR DESCRIPTION
**Pull request by ahiuchingau**
## overview
This PR adds robot rail lights control and allows monitoring of the robot light and door state to ProtocolContext, and bumps the max protocol API version to 2.5.

closes #5775 and closes #5703 

Note: GPIO control does not work on Jupyter Notebook

## changelog
- added the method `set_rail_lights()` and two properties `rail_lights_on` and `door_closed`
- updated docs to reflect the new changes

## risk assessment
Low

### CODE REVIEW

**Review by sfoster1 — APPROVED** [review]
LGTM! I think we should keep richer gpio interfaces on the table for future ux work though - allowing access to more pins, having interesting primitives like waiting for a pin to change, something like that.

**Comment by ahiuchingau** [comment]
@sfoster1 Totally agreed! Definitely something we should do in the near future

**Review by b-cooper — APPROVED** [review]
This is a great addition.  The documentation look thorough too 🍨 

I had a thought, I don't know if we document this elsewhere, but perhaps this would be a good place to have a `NOTE:` that explains that the top window and door must both be closed in order for the robot to report the door is closed? Just looking forward to the gotcha's the first time some one tries to use the door switch boolean without a top panel on their OT-2.

**Comment by SyntaxColoring** [comment]
> I had a thought, I don't know if we document this elsewhere, but perhaps this would be a good place to have a NOTE: that explains that the top window and door must both be closed in order for the robot to report the door is closed? Just looking forward to the gotcha's the first time some one tries to use the door switch boolean without a top panel on their OT-2.

Is that true? I didn't even realize the OT-2 had a way of detecting the presence of the top panel.

File: api/docs/v2/new_atomic_commands.rst [hunk 1]
```diff
@@ -502,3 +504,52 @@ The method :py:meth:`.ProtocolContext.comment` lets you display messages in the
...
+    protocol.rail_lights_on  # returns True when the lights are on,
+                             # False when the lights are off
+
+.. versionadded:: 2.5
+
+
+Monitor Robot Door
+==================
+
+The door safety switch feature flag has been added to the OT-2 software since the 3.19.0 release. Enabling the feature flag allows your robot to pause a running protocol and prohibit the protocol from running when the robot door is open.
```
Comment (Laura-Danielle) [inline api/docs/v2/new_atomic_commands.rst:545]: Maybe we can put a screenshot of where the feature flag is located? There are a few places that we have "feature flags" that might be confusing for the user.

File: api/docs/v2/versioning.rst [hunk 2]
```diff
@@ -142,3 +144,12 @@ Version 2.4
...
     - Takes into account geometry of the deck and modules
+
+
+Version 2.5
++++++++++++
+
+- New :ref:`new-utility-commands` were added:
+    - :py:meth:`.ProtocolContext.set_rail_lights`: turns robot rail lights on or off
+    - :py:attr:`.ProtocolContext.rail_lights_on`: describes whether or not the rail lights are on
+    - :py:attr:`.ProtocolContext.door_closed`: describes whether the robot door is closed
```
Comment (Laura-Danielle) [inline api/docs/v2/versioning.rst:155]: I'm a little worried about users trying to use the `door_closed` method + have the feature flag switched on.

Should we maybe put a note stating you should use one or the other to control robot behavior and/or only use this method to do a status check?

**Review by Laura-Danielle — CHANGES_REQUESTED** [review]

**Comment by ahiuchingau** [comment]
@b-cooper good call! I'll add it in!

@SyntaxColoring Yes - the top panel and the front panel switches are connected in series to the same GPIO pin. After OT-2 refresh, we will be able to identify which panel is open in new robots because they will be connected to separate pins.

**Review by Laura-Danielle — APPROVED** [review]
**N I C E**

**Review by sfoster1 — CHANGES_REQUESTED** [review]
Some interlinks in the docs don't quite work, they display in the interlink font-face but aren't actually links. According to [this random stackexchange answer](https://stackoverflow.com/questions/27170958/sphinx-documentation-how-to-reference-a-python-property) we may need to use `:py:obj:` to link to props across pages

**Review by sfoster1 — APPROVED** [review]
Nice! Thanks for fixing it.

**Pull request merged**